audiomentations/core/composition.py

```diff
@@ -125,20 +143,119 @@ class OneOf(BaseCompose):
+            )
+        return self.transform_indexes
+
+    def __call__(self, *args, **kwargs):
+        if not self.are_parameters_frozen:
+            kwargs["apply_to_children"] = False
+            self.randomize_parameters(*args, **kwargs)
+
+        if self.should_apply:
+            if "apply_to_children" in kwargs:
+                del kwargs["apply_to_children"]
+
+            if issubclass(type(self.transforms[0]), BaseSpectrogramTransform):
+                if "magnitude_spectrogram" in kwargs:
+                    transformed_data = self.transforms[self.transform_indexes[0]](
+                        kwargs["magnitude_spectrogram"]
+                    )
+                else:
+                    transformed_data = self.transforms[self.transform_indexes[0]](
+                        args[0]
+                    )
+
+                for transform_index in self.transform_indexes[1:]:
+                    transformed_data = self.transforms[
+                        self.transform_indexes[transform_index]
+                    ](transformed_data)
+
+            if issubclass(type(self.transforms[0]), BaseWaveformTransform):
+                # get access to the sample rate
+                if "sample_rate" in kwargs:
+                    samples = kwargs["samples"] if "samples" in kwargs else args[0]
+                    sample_rate = kwargs["sample_rate"]
+                else:
+                    samples = args[0]
+                    sample_rate = args[1]
+
+                transformed_data = self.transforms[self.transform_indexes[0]](
+                    samples, sample_rate
+                )
+
+                for transform_index in self.transform_indexes[1:]:
+                    transformed_data = self.transforms[transform_index](
+                        transformed_data, sample_rate
+                    )
+            return transformed_data
+
+        if "samples" in kwargs:
+            return kwargs["samples"]
+        elif "magnitude_spectrogram" in kwargs:
+            return kwargs["magnitude_spectrogram"]
+        else:
+            return args[0]
+
+
+class OneOf(BaseCompose):
+    """
+    OneOf randomly picks one of the given transforms when called, and applies that
+    transform.
+    Example usage:
+    ```
+    augment = OneOf([
+        TimeStretch(min_rate=0.8, max_rate=1.25, p=1.0),
+        PitchShift(min_semitones=-4, max_semitones=4, p=1.0),
+    ])
+    # Generate 2 seconds of dummy audio for the sake of example
+    samples = np.random.uniform(low=-0.2, high=0.2, size=(32000,)).astype(np.float32)
+    # Augment/transform/perturb the audio data
+    augmented_samples = augment(samples=samples, sample_rate=16000)
     # Result: The audio was either time-stretched or pitch-shifted, but not both
     ```
     """
 
     def __init__(self, transforms, p: float = 1.0):
         super().__init__(transforms, p)
-        self.transform_index = 0
+        self.transform_indexes = 0
```

Comment:
I feel like the plural here would imply that it's a list, but it's just an int, so I think `self.transform_index` works better here

Reply:
It was to have the same name for the two of them (more consistency)  but I agree with you that the singular is more appropriate. :)